oxd-server/src/main/java/org/gluu/oxd/server/op/GetTokensByCodeOperation.java

```diff
@@ -142,13 +144,19 @@ public IOpResponse execute(GetTokensByCodeParams params) throws Exception {
     }
 
     private void validate(GetTokensByCodeParams params) {
+
         if (Strings.isNullOrEmpty(params.getCode())) {
             throw new HttpException(ErrorResponseCode.BAD_REQUEST_NO_CODE);
         }
         if (Strings.isNullOrEmpty(params.getState())) {
             throw new HttpException(ErrorResponseCode.BAD_REQUEST_NO_STATE);
         }
-        if (!getStateService().isExpiredObjectPresent(params.getState())) {
+        try {
+            if (!getStateService().isExpiredObjectPresent(params.getState()) && !getStateService().isExpiredObjectPresent(Utils.decode(params.getState()))) {
```

Comment:
Put it inside `isExpiredObjectPresent`, so inside we check both encoded and decoded (it will cover other places where this method can be called).

Reply:
Made following implementation as discussed in chat. Removed `state` verification after decoding as it is now not required.

1) by default assume it’s already encoded and declare it in the documentation that it has to be encoded (means we don’t encode it by default)
2) add configuration option (with default value false), when admin changes it to true encode state value.